docs/first_semantic_game/02.simple_chat_npcs.rst

```diff
@@ -1,163 +1,395 @@
+
+The ``ReceiveMsg`` method is called when the NPC receives a message from another
+entity. It starts a new Task that calls ``ReceiveMsgAsync``. This is done so that
+the main thread can continue to run while the NPC is waiting for a response from
+GPT-4.
+
+.. code-block:: csharp
+
+    // Called when ChatEntity _inConvoWith emits a MsgSent signal
+    public override void ReceiveMsg(string msg)
+    {
+        // Call the async method but don't wait for it
+        Task.Run(() => ReceiveMsgAsync(msg));
+    }
+
+    public async Task ReceiveMsgAsync(string msg)
+    {
+        try
         {
-            Task.Run(() => ReplyAsync(msg));
+            // Add the message to the chat history.
+            _chat.AddUserMessage(msg);
+
+            // This is the call to GPT-4 to get the reply.
+            ChatRequestSettings settings = new();
+            string reply = await _chatGPT.GenerateMessageAsync(_chat, settings);
+
+            // Send the reply to the other entity via the MsgSent signal
+            SendMsg(reply);
+
+            // Add the reply to the chat history.
+            _chat.AddAssistantMessage(reply);
         }
-    
-        // Method to read, process, and respond to messages asynchronously
-        private async Task ReplyAsync(string msg)
+        catch (Exception ex)
         {
-            var chatHistory = _chatGPT.CreateNewChat();
-            chatHistory.AddUserMessage(msg);
-            var generatedMessage = await _chatGPT.GenerateMessageAsync(chatHistory);
-            EmitSignal(nameof(ReplyGenerated), generatedMessage);      
+            string errMsg = $"Could not get reply from {ChatName}: {ex.Message}";
+            GD.PrintErr(errMsg);
         }
     }
 
-Testing the Minimal `ChatAI` Class
-----------------------------------
 
-To verify the `ChatAI` minimal class, setup an interaction between two NPCs and allow them to communicate via `ReceiveMsg(msg)` method. Connect a handler function to the `ReplyGenerated` signal in order to observe the generated response, thus demonstrating an interactive conversation powered by Semantic Kernel.
+Two Arguing NPCs
+----------------
+
+In this section, you will learn how to set up NPCs named "East Troll" and 
+"West Troll" to automatically argue with each other using Godot's properties
+panel. These trolls are configured to have a debate about which side of the 
+river is better.
+
+Creating the East and West Troll Nodes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. **Create an NPC Scene**: Drag the ``NPC`` node from the main scene into 
+   the filesystem dock to create a new scene. This will convert the node
+   in the main scene to an instance of the new scene.
+2. **Duplicate the NPC**: Create a second troll by right-clicking the 
+   ``NPC.tscn`` scene in the filesystem dock and selecting "Duplicate". 
+3. **Rename Nodes**: Name the nodes ``West Troll`` the nodes ``East Troll``. Place the tokens on the left and right sides of the scene, respectively.

```

Comment:
I think the naming convention for nodes in Godot is PascalCase, so maybe change these to `WestTroll` and `EastTroll`. Would also have to change the MainGame.cs script, as noted.

Reply:
```suggestion
3. **Rename Nodes**: Name the nodes ``WestTroll`` the nodes ``EastTroll``. Place the tokens on the left and right sides of the scene, respectively.
```